includes/functions/functions_prices.php

```diff
@@ -985,29 +987,28 @@ function zen_get_products_sale_discount($product_id = false, $categories_id = fa
 // Actual Price Retail
 // Specials and Tax Included
   function zen_get_products_actual_price($products_id) {
-    global $db, $currencies;
+    global $db;
     $product_check = $db->Execute("select products_tax_class_id, products_price, products_priced_by_attribute, product_is_free, product_is_call from " . TABLE_PRODUCTS . " where products_id = '" . (int)$products_id . "'" . " limit 1");
 
-    $show_display_price = '';
-    $display_normal_price = zen_get_products_base_price($products_id);
-    $display_special_price = zen_get_products_special_price($products_id, true);
-    $display_sale_price = zen_get_products_special_price($products_id, false);
+    // If Free, Show it
+    if ($product_check->fields['product_is_free'] == '1') {
+      return 0;
+    }
 
-    $products_actual_price = $display_normal_price;
+    $display_sale_price = zen_get_products_special_price($products_id, false);
 
-    if ($display_special_price) {
-      $products_actual_price = $display_special_price;
-    }
     if ($display_sale_price) {
```

Comment:
Basically just from a code review, shouldn't this and the others be evaluated against `!== false` or something? Ie. what if there is a price of "0" intentionally returned as compared to the few cases in `zen_get_products_special_price` where `false` is returned and there are a few where `0` is returned (if calculated sale price is less than 0)?

Reply:
I suppose. For the last 10 years in production it hasn't been something complained about. Not that that's the only factor to consider. It just informs about the significance of the issue.

If one were to explicitly test for strict falsey values, what would the logic be? ie: what's the fallback action?

Reply:
Okay, I think you're right. Let's do the comparison `!== false` for safety.